src/TMedia.cpp

```diff
@@ -725,6 +728,28 @@ bool TMedia::doesMediaHavePriorityToPlay(TMediaData& mediaData, QString absolute
     return doesMediaHavePriorityToPlay;
 }
 
+void TMedia::checkMediaKeyConstraints(TMediaData& mediaData, QString absolutePathFileName)
+{
+    QList<TMediaPlayer*> mTMediaPlayerList = TMedia::getMediaPlayerList(mediaData);
+    QListIterator<TMediaPlayer*> itTMediaPlayer(mTMediaPlayerList);
+
+    while (itTMediaPlayer.hasNext()) {
+        TMediaPlayer* pTestPlayer = itTMediaPlayer.next();
+
+        if (pTestPlayer->getMediaPlayer()->state() == QMediaPlayer::PlayingState && pTestPlayer->getMediaPlayer()->mediaStatus() != QMediaPlayer::LoadingMedia) {
+            if (!mediaData.getMediaKey().isEmpty() && !pTestPlayer->getMediaData().getMediaKey().isEmpty()
+                && mediaData.getMediaKey() == pTestPlayer->getMediaData().getMediaKey()) { // Does it have the same key?
+                if (!pTestPlayer->getMediaPlayer()->media().canonicalUrl().toString().endsWith(absolutePathFileName)
+                    || (!mediaData.getMediaUrl().isEmpty() && !pTestPlayer->getMediaData().getMediaUrl().isEmpty()
+                        && mediaData.getMediaUrl() != pTestPlayer->getMediaData().getMediaUrl())) { // Is it a different sound or music than specified?
+                    TMediaData stopMediaData = pTestPlayer->getMediaData();
+                    mpHost->mpMedia->stopMedia(stopMediaData); // Stop it!
```

Comment:
This function does more than just check, it also acts - can I suggest a better name here?

Reply:
I updated the name to `matchMediaKeyAndStopMediaVariants` to provide more sense of the action.  The concept of the "key" will become clearer with a wiki update.